### PR TITLE
Connect to BookStore API

### PR DESCRIPTION
**Pull request by kaizipaul**
### These are the changes I made in this PR:
- Refactored my code to add and remove books through an API `bookstore api`
- Used `fetch` to send and receive data from the API
- Added new functionality to the components to display the books saved to the API.
- No styling was done.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,26 +1,101 @@
...
+    },
+  });
+  const books = await response.json();
+  const booksData = Object.keys(books).map((id) => ({
+    id,
+    title: books[id][0].title,
+    author: books[id][0].author,
+  }));
+  return booksData;
+};
```
Comment (EricMbouwe) [inline src/redux/books/books.js:26]: - [x] The project requires you to use `createAsyncThunk`  for creating thunk action creators. While what you implemented is working fine, you need to use the new redux approach, learn more about it [here](https://redux-toolkit.js.org/api/createAsyncThunk). Let's fix it 😀

Here is a hint for the `getBooks` action:

```js
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,26 +1,101 @@
...
+      title,
+      author,
+      category,
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  (await send.text());
+};
```
Comment (EricMbouwe) [inline src/redux/books/books.js]: - [x] As mentioned above you need to use `createAsyncThunk`  for creating thunk action creators. Let's fix that

File: src/redux/books/books.js [hunk 3]
```diff
@@ -1,26 +1,101 @@
...
+    method: 'DELETE',
+    body: JSON.stringify({
+      item_id: id,
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  (await remove.text());
+};
```
Comment (EricMbouwe) [inline src/redux/books/books.js]: - [x] As mentioned above you need to use `createAsyncThunk`  for creating thunk action creators. Let's fix that

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @kaizipaul , 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- Use fetch for API calls ✔
- Add book and remove book features working ✔
- No linter errors ✔

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @kaizipaul          , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x] Your project is professional 

- [x]  Your repo is professional 




 

- [x] No Linter Errors





Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**